### ISSUE TITLE
Rule idea: no-callback-literal

### ISSUE THREAD

**Issue by xjamundx** [eslint/eslint#6427]
I was wondering if we could have `no-callback-literal` rule to correspond with the `no-throw-literal` rule.

Basically, if you're using the callback pattern this should never be okay:

```js
callback('error')
```

instead you should do somethign more like this:

```js
callback(new Error('error'))
```

I already made this rule and it seems to work and could put it in a plugin if you don't want it.

**Comment by platinumazure** [comment]
How do we define a callback? Any function parameter we call as a function? Whitelist? How do we avoid collisions or unnecessary difficulty with rules like `callback-return`?

**Comment by xjamundx** [comment]
You just have to take the list of names. My rule uses `callback`, `cb` and `next`. Presumably it could be configurable if it had to be. I don't think it'd conflict with `callback-return` unless they changed the default callback options here: https://github.com/eslint/eslint/blob/master/lib/rules/callback-return.js#L27

**Comment by ilyavolodin** [comment]
@xjamundx How about a code like this:

```js
function foo(number, cb) {
  switch(number) {
    case 1: cb("one"); break;
    case 2: cb("two"); break;
    ...
  }
}
```

Seems valid to me. I see your point that in most cases it's makes no sense to return string literal to callback, but there are valid edge-cases.

**Comment by xjamundx** [comment]
@ilyavolodin sure it's an exception. That code doesn't follow the standard callback pattern.

**Comment by ilyavolodin** [comment]
I'm not against it, but I think it's too situational for inclusion into the core.

**Comment by xjamundx** [comment]
Any plugins you think might want something like this?

**Comment by ilyavolodin** [comment]
Hmm.. No idea, sorry. Let's see what the rest of the team thinks. If others feel strongly that it should be included, I'm not opposed.

**Comment by mysticatea** [comment]
Though it's trivial, just I felt `null` also is a literal, but `null` is valid here. 😜
If it's not core, eslint-plugin-node might be properly.
I don't have strong opinions.

**Comment by xjamundx** [comment]
Will open this at `eslint-plugin-standard` for now, thanks!

**Comment by Zyles** [comment]
Okay....

And how do you disable this rule, because there is no such rule...
